f8a_worker/process.py

```diff
@@ -440,13 +454,20 @@ def fetch_rubygems_artifact(name, version, target_dir):
 
     @staticmethod
     def fetch_scm_artifact(name, version, target_dir):
-        git_url = 'git://' + name
-        git = Git.clone(git_url, target_dir, single_branch=True)
-        git.reset(version, hard=True)
-        filename = git.archive(version)
-        artifact_path = os.path.join(target_dir, filename)
-        digest = compute_digest(artifact_path)
-        return digest, artifact_path
+        env = dict(os.environ)
+        env['GOPATH'] = target_dir
+        TimedCommand.get_command_output(['go', 'get', '-d', name],
+                                        timeout=300,
+                                        env=env,
+                                        graceful=True)
+        package_dir = os.path.join(target_dir, 'src', name)
+        with cwd(package_dir):
+            git = Git(package_dir)
+            git.reset(version, hard=True)
+            artifact_filename = git.archive(version, repo_path='.')
```

Comment:
I'd guess that `repo_path='.'` does the same as not specifying `repo_path` at all, doesn't it ?

Reply:
@jpopelka The default value of `repo_path` parameter is `None`. Not sure if `with cwd(self.repo_path)` will fail with that value. @msrb: Could you comment on this ?

Reply:
It won't fail because there's `repo_path` argument and `self.repo_path` attribute.
Those two are different things, hence I'd suggest renaming `repo_path` argument.

Reply:
Ah, good catch. @jpopelka is right. `repo_path` from `archive()` method overshadows `self.repo_path` object attribute. It works, but it's confusing. We should rename the method param.